src/proxyprotocol/address.go

```diff
@@ -0,0 +1,57 @@
+package proxyprotocol
+
+import (
+	"net"
+	"strings"
+
+	proxyproto "github.com/pires/go-proxyproto"
+)
+
+// NewProxyAddr creates an Addr struct from supplied
+// proxyproto.AddressFamilyAndProtocol, net.IP and port as uint16.
+func NewProxyAddr(proto proxyproto.AddressFamilyAndProtocol, addr net.IP, port uint16) net.Addr {
+	network := convertProxyProtocolToString(proto)
+	switch strings.ToLower(network) {
+	case "unix", "unixstream", "unixdgram":
+		return &net.UnixAddr{
+			Net:  network,
+			Name: addr.String(),
+		}
+	case "udp", "udp4", "udp6":
+		return &net.UDPAddr{
+			IP:   addr,
+			Port: int(port),
+		}
+	default:
+		return &net.TCPAddr{
+			IP:   addr,
+			Port: int(port),
+		}
+	}
+}
+
+func convertProxyProtocolToString(afp proxyproto.AddressFamilyAndProtocol) string {
+	if afp.IsIPv4() {
+		// IPv4
```

Comment:
These comments add nothing

Reply:
I originally had comments the whole way down but they got removed when cleaning it up, removing them now.